technical_standards/testing/strategy.md

```diff
@@ -0,0 +1,96 @@
+
+![](./_images/diamont.png)
+
+### Acceptance criteria are the base
+
+When testing, it is not possible to test everything, and there is always a moment we need to stop.
+
+To create a clear limit, we decided to use acceptance criteria to guide our tests.
+
+Acceptance criteria are a couple of sentences each one describing an atomic behavior from the application that should be met once the task is done.
+
+This way it is now possible for the developer to easily check the changes are following requirements without having any assistance from the person who created the task.
+
+This can seem a bit overkill. However, often tasks can stack up in the backlog for weeks if it is not months or years. Having the done description in this case is even more useful as in this case the person who created the task might have forgot what was the initial request.
+
+```mermaid
+sequenceDiagram
+    actor Product owner
+    participant Task
+    actor Developer
+    participant Test
+    participant Code
+    
+    Developer ->> Task: Check the task
+    Developer ->> Product owner: Check requirements
+    Developer ->> Test: Create a test based on the requirements
+    Developer ->> Test: Check sure test fails
+    alt test succeed
+        Developer ->> Product owner: Notify the issue is potentially fixed
+    else
+        Developer ->> Code: Implement new code so the test is succeeding
+    end
+        Developer ->> Task: Mark the task as done 
+```
+**Without Acceptance criteria**
+
+```mermaid
+sequenceDiagram
+    actor Product owner
+    participant Task
+    actor Developer
+    participant Test
+    participant Code
+    
+    Product owner ->> Task: Create acceptance criteria based on business logic
+    Developer ->> Task: Check acceptance criteria
+    Developer ->> Test: Create a test based on the acceptance criteria
+    Developer ->> Test: Check sure test fails
+    alt test succeed
+        Developer ->> Product owner: Notify the issue is potentially fixed
+    else
+        Developer ->> Code: Implement new code so the test is succeeding
+    end
+        Developer ->> Task: Mark the task as done 
+```
+**With Acceptance criteria**
+
+To create a valid acceptance criteria, we need always to have at least three parts respecting the AAA framework:
+- **Arrange**: In the first part of the acceptance criteria, we need to define the initial state.
+- **Act**: This part is here to describe the part of the code that will be executed and so tested.
+- **Assert**: The goal from that last part is to make sure that we are in the state we expect.
+
+![Acceptance criteria](./_images/AC.png)
+
+### The best code is no code
+
+Each line of code ever wrote will add up to the legacy that you will have to maintain.
+
+This is the reason why we always want to write our test before changing any line of code inside the application.
```

Comment:
This paragrpah is not very actionable on its own. Not everyone uses TDD and this is not something we enforce. So I am not sure about how this paragprah brings value?

Reply:
I agree with your comment. To clarify mine, I might have wrongly used the word TDD. I mostly agree with the guideline you suggest, but I don't think this text is teaching anything or providing value. Also, "writing tests first" is not an enforced practice in the team, so, at this stage, the combination of:
1. Not actionable guideline ;
2. Not commonly enforced practice in the team
makes the handbook a bit disconnected from what the team does, without an easy way to get closer to it.

I would recommend starting first with gathering commonly accepted and used practices in the team.